aten/src/ATen/native/xnnpack/Common.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <ATen/ATen.h>
+
+#ifdef USE_XNNPACK
+
+#include <xnnpack.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+namespace internal {
+
+struct Layout final {
+  // 4D Activation Maps
+  struct Activation4D final {
+    static constexpr size_t Batch = 0u;
+    static constexpr size_t Channels = 1u;
+    static constexpr size_t Height = 2u;
+    static constexpr size_t Width = 3u;
+  };
+
+  // ND Activation Maps
+  struct ActivationND final {
+    // Some operators may not be limited to 4 dimensional tensors. In that scenario,
+    // XNNPACK denotes that operator with an _nc suffix and expects all dimensions,
+    // except channels, to be flattened into one argument: batch_size.
+    static int64_t Batch(const IntArrayRef tensor) {
```

Comment:
I don't think any other place in Pytorch uses capitalized function names.  (Capitals are only used for types, AFAIK.)  Any particular reason this is named like this, not `batch`?

Reply:
Also, it is somewhat confusing that Activation4d::Batch is a constexpr that gives the index of a batch dimension in a tensor, so to get actual number of batches one would do `x.size(Activation4d::Batch)`, but ActivationNd::Batch is a function that gives the number of batches (redefined to suit its purpose, but that's a separate issue), so to get number of batches one would do `ActivationNd::Batch(x.sizes())`